docs/pages/store/encoding.mdx

```diff
@@ -7,381 +7,427 @@ It is comparable to `abi.encodePacked`, but with some notable differences:
 
 - **Array elements are tightly packed**, without any padding.
   This might cause some elements to wrap around two storage slots, but saves a lot of space.
-  For example, an `address[3]` array will only use 2 storage slots in MUD instead of 3 in Solidity,
-  leading to a 33% reduction in storage costs.
+  For example, an `address[3]` array will only use 2 storage slots in MUD instead of 3 in Solidity, leading to a 33% reduction in storage costs.
 - **Array lengths are packed into a single slot**.
-  Since table are limited to up to 5 dynamic field, and the length of each field is limited to `2**40` (~1 trillion),
-  we can pack the lengths of all dynamic length fields of a table into a single storage slot.
-  Compared to Solidity, this saves ~1 storage slot per dynamic length element.
+  Tables are limited to up to 5 dynamic fields, and the length of each field is limited to 2<sup>40</sup> (over a million millions), we can pack the lengths of all dynamic length fields of a table into a single storage slot.
+  Compared to Solidity, this saves 1 storage slot per dynamic length element except for the first one.
 
 This encoding scheme greatly reduces gas for storage operations on dynamic length fields and for emitting events.
-**For events it reduces the payload size by ~80%** compared to the Solidity default of `abi.encode` (depending on the table schema).
+**For events it reduces the payload size by over 80%** compared to the Solidity default of `abi.encode` (depending on the table schema).
 Note that since the encoding happens at runtime and does not have access to compiler internals, there is some overhead for encoding and decoding that is not present in vanilla Solidity.
 
-## Schema
+## Schemas
 
-Each table has a key schema and a value schema, which define the data types of each column.
+Each table has two schemas: a key schema and a value schema.
+The schemas define the data type of each column.
```

Comment:
This paragraph is a but confusing with the new config where the schema is defined as one schema and some columns are marked as key. Onchain we still distinguish between key schema and value schema though. How can we rephrase this to make it more clear? One schema but it's stored separately for value fields and key fields onchain (because onchain the table is more like a key/value store than a table, ie we can't look up by value, only by key)

Reply:
Can you check the new explanation? I think it's clearer.